evap/student/views.py

```diff
@@ -105,12 +105,51 @@ def sorter(evaluation):
         )
 
     unfinished_evaluations.sort(key=sorter)
+    info = Evaluation.annotate_with_participant_and_voter_counts(
+        Semester.active_semester()
+        .evaluations.filter(is_single_result=False)
+        .exclude(state__lt=Evaluation.State.APPROVED)
+        .exclude(is_rewarded=False)
+        .exclude(id__in=settings.EXCLUDED_EVALUATION_IDS)
+        .exclude(course__type__id__in=settings.EXCLUDED_COURSE_TYPE_IDS)
+        .exclude(course__is_private=True)
+    ).aggregate(
+        total_votes=Sum("num_voters"),
+        participants=Sum("num_participants"),
+    )
+    if settings.GLOBAL_EVALUATION_PROGRESS_REWARDS:
+        max_reward = max(settings.GLOBAL_EVALUATION_PROGRESS_REWARDS)
+        final_goal = round(info["participants"] * max_reward[0])
+        next_goal = final_goal
+        next_reward = max_reward[1]
+        for percentage, reward in settings.GLOBAL_EVALUATION_PROGRESS_REWARDS:
+            candidate = info["participants"] * percentage
+            if info["total_votes"] < candidate < next_goal:
+                next_goal = round(candidate)
+                next_reward = reward
```

Comment:
I think this would benefit from a bit more functional approach, something like
```python3
next_reward = min(
    reward for reward in settings.GLOBAL_EVALUATION_PROGRESS_REWARDS
    if current_participation_ratio < reward.required_participation_ratio
)
```